docs/app/js/app.js

```diff
@@ -881,4 +882,37 @@ function($rootScope, $scope, component, demos, $templateRequest) {
       }
     }
   };
-});
+})
+
+/** This directive caches the scroll position for each route + templateUrl combination
+ * This helps in restoring the scroll when the user visits the same page again
+ */
+.directive('cacheScrollPosition', ['$route', '$mdUtil', '$timeout', '$location', '$anchorScroll',
+function($route, $mdUtil, $timeout, $location, $anchorScroll) {
+  var scrollPosCache = {};
+  var mainContentArea = document.querySelector("[role='main']");
+  var scrollContentEl = mainContentArea.querySelector('md-content[md-scroll-y]');
+  return function(scope) {
+
+      scope.$on('$routeChangeStart', function() {
```

Comment:
Please reformat this new block to use 2 space indentation.

Reply:
You probably want to run `npm run lint` to check for any other issues as well if you don't have ESLint enabled in your IDE.

Reply:
Sure, I do have eslint enabled in my editor and it warns me on other rules, but not in space/tabs. Anyway would run limiting before commit.